src/v/security/authorizer.cc

```diff
@@ -72,6 +143,20 @@ const acl_store& authorizer::store() const& { return *_store; }
 
 template<typename T>
 auth_result authorizer::authorized(
+  const T& resource_name,
+  acl_operation operation,
+  const acl_principal& principal,
+  const acl_host& host) const {
+    auth_result r = do_authorized(resource_name, operation, principal, host);
+    _probe->record_authz_result(
+      r.empty_matches     ? authz_result::empty
+      : r.is_authorized() ? authz_result::allow
+                          : authz_result::deny);
```

Comment:
Maybe this feels slightly safe, but I'm fairly sure it's the same thing:
```suggestion
      r.is_authorized() ? authz_result::allow
      : r.empty_matches ? authz_result::empty
                        : authz_result::deny);
```

Reply:
what you have seems fine. I'd prefer to elide the wacky "allow empty matches" behavior as much as possible.

Reply:
> what you have seems fine. I'd prefer to elide the wacky "allow empty matches" behavior as much as possible.

I could just record allow/deny, but I feel like empty is sort of useful?

Reply:
yeah, sorry. i agree empty is a useful distinction. 

I just mean the second version (your `suggestion` diff) accounts for the case where `is_authorized() && empty_matches`, which can only occur in tests. nothing to do here imo.